apps/native/app/src/screens/license-scanner/license-scan-detail.tsx

```diff
@@ -1,131 +1,56 @@
-import { ScanResultCard } from '@ui'
-import { BarCodeEvent, Constants } from 'expo-barcode-scanner'
-import React, { useEffect, useState } from 'react'
+import { ScanResultCard, SupportedGenericLicenseTypes } from '@ui'
 import { useIntl } from 'react-intl'
-import { View } from 'react-native'
 import {
   Navigation,
   NavigationFunctionComponent,
 } from 'react-native-navigation'
 import { useNavigationButtonPress } from 'react-native-navigation-hooks/dist'
+import styled from 'styled-components/native'
+import {
+  VerifyLicenseBarcodeError,
+  VerifyLicenseBarcodeMutation,
+} from '../../graphql/types/schema'
 import { StackRegistry } from '../../utils/component-registry'
-import { LicenseScanResult } from './scan-results/license-scan-result'
-
-enum ScanResult {
-  DRIVER_LICENCE = 'DriversLicense',
-  FIREARM_LICENSE = 'FirearmLicense',
-  ADR_LICENSE = 'AdrLicense',
-  MACHINE_LICENSE = 'MachineLicense',
-  DISABILITY_LICENSE = 'DisabilityLicense',
-  HUNTING_LICENSE = 'HuntingLicense',
-  UNKNOWN = 'Unknown',
-}
-
-const DriversLicenseTemplateIds = [
-  'cd998d1b-3cfa-4753-9f1e-6dc5187c59f9',
-  '3d2a9e02-24ef-446b-ab4a-f34b26850460',
-]
 
-const FirearmTemplateIds = [
-  'dfb706c1-3a78-4518-bf25-cebbf0a93132',
-  '61f74977-0e81-4786-94df-6b8470013f09',
-]
-const VVRTemplateIds = [
-  '61012578-c2a0-489e-8dbd-3df5b3e538ea',
-  'd63e4b27-9a0b-45b1-8fe4-a9be0011781f',
-]
-const ADRTemplateIds = [
-  '4e49febe-7ca9-49e3-a3be-3be70cb996c2',
-  '62a105d5-ef8a-419e-8423-fbbbfb5f1b36',
-]
-const DisabilityTemplateIds = [
-  'd0b1a6b6-3af3-4131-9b97-c6d7d2ed7e63',
-  'c78364b6-33a8-4242-84ca-24de0854fe00',
-]
+const Host = styled.View`
+  flex: 1;
+  padding: ${({ theme }) => theme.spacing[2]}px;
+`
 
-const HuntingLicenseTemplateIds = [
-  '1da72d52-a93a-4d0f-8463-1933a2bd210b',
-  '5f42f942-d8d6-40bf-a186-5a9e12619d9f',
-]
+interface LicenseScanDetailScreenProps {
+  verifyLicenseBarcode: VerifyLicenseBarcodeMutation['verifyLicenseBarcode']
+}
 
 export const LicenseScanDetailScreen: NavigationFunctionComponent<
-  BarCodeEvent & { isExpired: boolean }
-> = ({ data, type, isExpired }) => {
-  const [loaded, setLoaded] = useState(false)
-  const [scanResult, setScanResult] = useState<ScanResult>(ScanResult.UNKNOWN)
+  LicenseScanDetailScreenProps
+> = ({ verifyLicenseBarcode: { licenseType, data, error } }) => {
   const intl = useIntl()
 
+  // We need to cast licenseType to SupportedGenericLicenseTypes because not all GenericLicenseTypes are supported
+  const type = licenseType as unknown as SupportedGenericLicenseTypes
+  const isExpired = error === VerifyLicenseBarcodeError.Expired
+
   useNavigationButtonPress(({ buttonId }) => {
     if (buttonId === 'LICENSE_SCANNER_DONE') {
       Navigation.dismissModal(StackRegistry.LicenseScannerStack)
```

Comment:
Should we remove done since we already have a back button which does exactly the same?